src/test/java/io/cdap/plugin/sfmc/sink/DataExtensionRecordWriterTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.sfmc.sink;
+
+import com.exacttarget.fuelsdk.ETDataExtensionColumn;
+import com.exacttarget.fuelsdk.ETDataExtensionRow;
+import com.exacttarget.fuelsdk.ETResponse;
+import com.exacttarget.fuelsdk.ETResult;
+import com.exacttarget.fuelsdk.ETSdkException;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.junit.Test;
+import org.mockito.Mockito;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataExtensionRecordWriterTest {
+
+  @Test
+  public void testWriteWInsertWNullResponse() throws ETSdkException, IOException {
+    StructuredRecord record = Mockito.mock(StructuredRecord.class);
+    List<ETDataExtensionRow> batch = new ArrayList<>();
+    ETDataExtensionRow row1 = new ETDataExtensionRow();
+    row1.setId("121");
+    row1.setDataExtensionKey("DE");
+    ETDataExtensionRow row2 = new ETDataExtensionRow();
+    row1.setId("122");
+    batch.add(row1);
+    batch.add(row2);
+    batch.size();
+    List<ETDataExtensionColumn> columnList = new ArrayList<>();
+    ETDataExtensionColumn column = new ETDataExtensionColumn();
+    column.setLength(123);
+    column.setName("Name");
+    columnList.add(column);
+    DataExtensionClient dataExtensionClient = Mockito.mock(DataExtensionClient.class);
+    DataExtensionInfo info = new DataExtensionInfo("externalKey", columnList);
+    RecordDataExtensionRowConverter recordDataExtensionRowConverter = new RecordDataExtensionRowConverter(info, true);
+    DataExtensionRecordWriter dataExtensionRecordWriter = new DataExtensionRecordWriter(dataExtensionClient,
+      recordDataExtensionRowConverter, Operation.INSERT, 1, false);
+    dataExtensionRecordWriter.write(null, record);
```

Comment:
looks like you're not validating anything here

Reply:
Added the validation.

Reply:
you're calling dataExtensionRecordWriter.write(null, record) in the test code, and then verifying that dataExtensionRecordWriter.write(null, record) was called once. That's not really validating anything meaninful :)
Same comment in other tests

Reply:
These test cases are basically for writeBatch method which had a private access but now it is made package private and the real call for it has been added in verify by getting it invoked by write() or close() methods.